### ISSUE TITLE
Indication of sticky/featured content for cards

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#6695]
# Story

As a website visitor, I can tell what content has been marked as sticky/featured both visually and with a screen reader..

# Proposed solution

- [x] Be aware of where sticky is available
- [x] Implement new pre-title slot in card in uids for icon and pinned text in https://github.com/uiowa/uiowa/issues/6695#issuecomment-2389530290
- [x] Update the card teaser to indicate what content has been marked as sticky and is not following the normal sort.
- [x] Make sure non-visual users can get the same information when traversing the content.

**Comment by pyrello** [comment]
Thanks for putting this together, @joewhitsitt!

**Comment by bspeare** [comment]
For the visual display of sticky items, perhaps we would increase the title and teaser text of the card to `font-size: 125%;` and separate it from the rest of the list/grid items with a horizontal rule?   I've added some examples of how it could look with the list view and grid view below. 

<img width="1552" alt="Screenshot 2024-09-20 at 9 32 00 AM" src="https://github.com/user-attachments/assets/33c344e2-b573-4f66-9b0a-0a376bc968c8">

<img width="1575" alt="Screenshot 2024-09-20 at 9 33 15 AM" src="https://github.com/user-attachments/assets/6171afea-7155-4496-a14c-afbfee1bafc4">

**Comment by joewhitsitt** [comment]
So if there were multiple, would they each get a hr or just after the last one? I could see folks unintentionally having more than one page of sticky and they will be like, "why isn't this in a grid like I set it"

**Comment by joewhitsitt** [comment]
then what about combined with media images, sticky and grid. we would have to call a separate view mode for the sticky so that the image is right or left instead of stacked?

**Comment by bspeare** [comment]
> So if there were multiple, would they each get a hr or just after the last one? I could see folks unintentionally having more than one page of sticky and they will be like, "why isn't this in a grid like I set it"

Yup, each one would get an hr.

**Comment by bspeare** [comment]
> then what about combined with media images, sticky and grid. we would have to call a separate view mode for the sticky so that the image is right or left instead of stacked?

We certainly could do something like that.  I'm just looking for a minimal way to highlight sticky content yet still let the display options control the card styles.

**Comment by joewhitsitt** [comment]
I guess I am getting at that it doesn't seem very minimal when we have to deal with a full width card that was otherwise set with grid/stacked.

We have previously mentioned something like a pin icon with an aria-label. To me that seems like a small slice given that we could just preprocess teaser titles to prepend that regardless of what card styles are set on the block. I'd be interested in hearing concerns about that.

**Comment by bspeare** [comment]
Some sort of pin icon sounds good to me!

<img width="1411" alt="Screenshot 2024-09-20 at 10 39 46 AM" src="https://github.com/user-attachments/assets/34105b04-9893-4e5f-ae19-9e981fface21">

<img width="1439" alt="Screenshot 2024-09-20 at 10 41 32 AM" src="https://github.com/user-attachments/assets/5fdc65d6-9509-4304-89c4-ed515bb82ba5">

**Comment by pyrello** [comment]
> Some sort of pin icon sounds good to me!
> 
> <img alt="Screenshot 2024-09-20 at 10 39 46 AM" width="1411" src="https://private-user-images.githubusercontent.com/1036433/369467142-34105b04-9893-4e5f-ae19-9e981fface21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3MTE5ODAsIm5iZiI6MTcyNzcxMTY4MCwicGF0aCI6Ii8xMDM2NDMzLzM2OTQ2NzE0Mi0zNDEwNWIwNC05ODkzLTRlNWYtYWUxOS05ZTk4MWZmYWNlMjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MzBUMTU1NDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjBlNDJkZDJiY2JjM2I2YjIwNjMzNTVjNjJiODAyY2Q4MGNiZmVmZjBhMDc5YzEzOGMzYWI0NjAwZTlmMDU3YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.T_TqjqM-VVO0nTAiZXczofMC3CbRlpiDwzf7_XyRjC4"> <img alt="Screenshot 2024-09-20 at 10 41 32 AM" width="1439" src="https://private-user-images.githubusercontent.com/1036433/369467625-5fdc65d6-9509-4304-89c4-ed515bb82ba5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3MTE5ODAsIm5iZiI6MTcyNzcxMTY4MCwicGF0aCI6Ii8xMDM2NDMzLzM2OTQ2NzYyNS01ZmRjNjVkNi05NTA5LTQzMDQtODljNC1lZDUxNWJiODJiYTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MzBUMTU1NDQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjMzOWI2MDNjYTgzZjUyMGE5OGVmYjAyNWUxNTE3N2E3ZDdjNTI5MjQ5OWM4YWQzMmM3NTFkNWQ0NWFmOTUxYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.DbjQf-vi0GFJxXfOUEq83mddd7wJSETU38l1bAg1kI0">

I'm wondering if we could have the pin icon not disrupt the display of the item and have it to the left of the item in the margin.

**Comment by briand44** [comment]
grooming conversation... May need to do something in UIDS depending on how we determine to move forward with this. We may need to audit to see what besides cards would need accounted for.

**Comment by bspeare** [comment]
Adding it to the end of the title seems the least disruptive. 

![Untitled-1-06](https://github.com/user-attachments/assets/c3a7ba66-ebe8-45a3-8df4-1ae1e1469549)
![Untitled-1-05](https://github.com/user-attachments/assets/6b9882f2-c38f-46a9-82b5-6291a401e7fd)

Or adding it above and outside the card would also avoid disrupting the card markup. 

![Untitled-1-06](https://github.com/user-attachments/assets/246255ed-ae91-4ed7-8ed9-7d8c83b87a88)
![Untitled-1-05](https://github.com/user-attachments/assets/72b31c7e-9637-4a1f-8bd8-f509cf71a20d)


A corner badge would be cool, but would be difficult since the card does not have padding in some variants.
 
![Untitled-1-03](https://github.com/user-attachments/assets/527fe9a1-1a30-407a-9c9d-fd95922a8a83)

**Comment by joewhitsitt** [comment]
oooh! i like end of title black and yellow icon. poor-man's duotone fontawesome.

the grey "pinned" looks nice but it seems disconnected from the content when there is a border.

The triangle reminds me too much of the old Tippie site and sounds the most complicated.

**Comment by bspeare** [comment]
@joewhitsitt we could do a stack, which is free!   https://docs.fontawesome.com/web/style/stack

**Comment by bspeare** [comment]
We could also move the gray "pinned" option inside the card to like a pre_title slot.
![Untitled-1-06](https://github.com/user-attachments/assets/4241c921-f52f-4ed3-88f6-4506f566dd90)
![Untitled-1-05](https://github.com/user-attachments/assets/776afb1d-f2f8-4302-b260-7de169c400c7)

**Comment by pyrello** [comment]
> We could also move the gray "pinned" option inside the card to like a pre_title slot. 

I think this is my fav option so far.

**Comment by quamsta** [comment]
Can anything go in the upper right? 

<img width="1024" alt="Screenshot 2024-10-03 at 10 45 47 AM" src="https://github.com/user-attachments/assets/6abca61b-e34e-4e61-a669-ee0d79bc5fb4">

Doesn't work as well with a borderless card though I guess.

**Comment by briand44** [comment]
Grooming conversation... Discussed moving forward with this work for cards. If we want to address other instances not using card that can be a later slice and may require the audit in #8124 . Sounded like we would be moving forward with the "Pinned" example from https://github.com/uiowa/uiowa/issues/6695#issuecomment-2389530290 and we can get feedback in an EOD. Sized as large since scope includes both UIDS and uids_base work.

**Comment by joewhitsitt** [comment]
should the storybook 8 PR be merged first?

**Comment by joewhitsitt** [comment]
Started https://github.com/uiowa/uids/pull/940

**Comment by joewhitsitt** [comment]
>Be aware of where sticky is available

https://github.com/uiowa/uiowa/pull/8199

Curious about my approach here. Since the UIDS addition is only scoped to cards, I am checking to see if the content is part of a view that has sticky sorting before checking to see if the content itself is sticky. I am doing this in the docroot/modules/custom/uiowa_core/src/Entity/NodeBundleBase.php to target all node content at the right time (before the card is built) with the right information (view/sticky).

**Comment by joewhitsitt** [comment]
Posting a comment from Sean (Slack DM):

>I think this looks good. I can't say for sure we might not eventually refactor it, but I generally like the premise of that the logic lives in the NodeBundleBase class.

**Comment by bewoldt** [comment]
Will there be a way to hide this? We use the sticky option a lot on Staff pages to position the Director first. I just don't think the "pinned" terminology fits when dealing with staff.

**Comment by joewhitsitt** [comment]
Interesting point. I think this stemmed though from issues where it wasn't clear how content was sorted and bringing transparency and consistency to that, so I am not sure hiding it on a case-by-case or site-by-site basis would be the solution here.

I think there are other ways to achieve your custom sorting without using the sticky feature. Featured content block, tagging/filtering people into separate blocks with identifying headers.

Perhaps this could be solved using a different term based on the content type being shown. Pinned by default, overrides for content like people?

**Comment by bewoldt** [comment]
Most of those options don't work for most scenarios with people lists. I'm surprised people lists weren't mentioned in this issue. Every example reference so far in this issue uses articles so why not just only show this on article lists and leave everything else off?

**Comment by cory-skeers** [comment]
> Most of those options don't work for most scenarios with people lists. I'm surprised people lists weren't mentioned in this issue. Every example reference so far in this issue uses articles so why not just only show this on article lists and leave everything else off?

There are other examples such as alerts or events where sticky is used and the pinned display/label is appropriate, so it can't be limited to only articles.

We can look at making exceptions or other options with people though, or if there are other lists where that would be appropriate.

**Comment by quamsta** [comment]
Yeah, I agree with @bewoldt  here. While you can use featured content, etc to manually order people, you lose a lot of formatting/image options that people lists have. The fact that a user can go through the steps of tagging people, making a list block of people, sticky an individual, and they show up labeled as "Pinned" just seems odd. I'd say an exception for people is warranted here.

In my opinion, if a person is listed first on a page, and they have a title like "Director" or something similar, the context is clear why they're listed first. Articles, pages, etc don't have that sort of baked-in context in terms of content hierarchy.
